### PR TITLE
Don't allow executing archived pivot-table cards in a public dashboard

### PR DESCRIPTION
**Pull request by calherries**
This PR fixes a bug where it is possible to execute a pivot-table card that is archived in a public dashboard from an API endpoint.

### CODE REVIEW

File: src/metabase/api/public.clj [hunk 1]
```diff
@@ -606,6 +606,7 @@
    dashcard-id ms/PositiveInt
    parameters  [:maybe ms/JSONString]}
   (validation/check-public-sharing-enabled)
+  (api/check-404 (t2/select-one-pk :model/Card :id card-id :archived false))
```
Comment (calherries) [inline src/metabase/api/public.clj:609]: This check was missing.

See the similar check here for non-pivot-table cards:
https://github.com/metabase/metabase/blob/f52e15255b3b26635605ffa203a1cd6a4e0635e5/src/metabase/api/public.clj#L287

File: test/metabase/api/public_test.clj [hunk 2]
```diff
@@ -1550,56 +1550,90 @@
   [dash card dashcard]
   (format "public/pivot/dashboard/%s/dashcard/%d/card/%d" (:public_uuid dash) (u/the-id dashcard) (u/the-id card)))
 
+(defn- do-with-temp-dashboard-and-public-pivot-card [f]
```
Comment (calherries) [inline test/metabase/api/public_test.clj:1553]: I extracted this to reuse it in multiple tests

File: test/metabase/api/public_test.clj [hunk 3]
```diff
@@ -1550,56 +1550,90 @@
...
+                 (testing "row_count isn't included in public endpoints"
+                   (is (nil? (:row_count result))))
+                 (is (= 6 (count (get-in result [:data :cols]))))
+                 (let [rows (mt/rows result)]
+                   (is (= 80 (count rows)))
+                   (is (= ["CA" "Affiliate" "Doohickey" 0 16 48] (first rows)))
+                   (is (= [nil "Google" "Gizmo" 1 52 186] (nth rows 50)))
+                   (is (= [nil nil nil 7 1015 3758] (last rows)))))))))))))
+
+(deftest public-pivot-dashcard-errors-test
```
Comment (calherries) [inline test/metabase/api/public_test.clj:1609]: This new test mirrors the existing one for non-pivot-table cards:
https://github.com/metabase/metabase/blob/fd6f5fad7ca5ac449c1ac20dff09335c270c37ce/test/metabase/api/public_test.clj#L617-L644

File: test/metabase/api/public_test.clj [hunk 4]
```diff
@@ -1550,56 +1550,90 @@
...
+  (testing "GET /api/public/pivot/dashboard/:uuid/dashcard/:dashcard-id/card/:card-id"
+    (testing "Shouldn't be able to execute a public DashCard if public sharing is disabled"
+      (mt/test-drivers (api.pivots/applicable-drivers)
+        (do-with-temp-dashboard-and-public-pivot-card
+         (fn [dash card dashcard]
+           (mt/with-temporary-setting-values [enable-public-sharing false]
+             (is (= "An error occurred."
```
Comment (crisptrutski) [inline test/metabase/api/public_test.clj:1616]: A testing label would be nice here

**Review by crisptrutski — APPROVED** [review]

File: test/metabase/api/public_test.clj (hunk 4 above)
Comment (calherries) [reply test/metabase/api/public_test.clj:1616]: Good spotting, I rearranged some things and forgot to move the testing description. Updated in 19d280ece8f619c8992e2911b50a19c7a38f620e

**Pull request merged**